src/thor/costmatrix.cc

```diff
@@ -190,7 +193,7 @@ bool CostMatrix::SourceToTarget(Api& request,
               if (targets.empty() && locs_status_[MATRIX_FORW][source].threshold > 0) {
                 // TODO(nils): shouldn't we extend the search here similar to bidir A*
                 //   i.e. if pruning was disabled we extend the search in the other direction
-                locs_status_[MATRIX_FORW][i].threshold = -1;
+                locs_status_[MATRIX_FORW][source].threshold = -1;
```

Comment:
this was a bug, I wonder how it never segfaulted!

Reply:
the second part of the condition must be very uncommon? code coverage says none of our tests make it happen either